nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/remote/StandardRemoteProcessGroup.java

```diff
@@ -856,10 +864,75 @@ public void refreshFlowContents() throws CommunicationsException {
         }
     }
 
+    @Override
+    public String getNetworkInterface() {
+        readLock.lock();
+        try {
+            return networkInterfaceName;
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public void setNetworkInterface(final String interfaceName) {
+        writeLock.lock();
+        try {
+            this.networkInterfaceName = interfaceName;
+
+            try {
+                final Enumeration<InetAddress> inetAddresses = NetworkInterface.getByName(interfaceName).getInetAddresses();
+
+                if (inetAddresses.hasMoreElements()) {
+                    this.localAddress = inetAddresses.nextElement();
```

Comment:
@phrocker you are right - the user is selecting a name that could potentially map to multiple IP's. I'm not sure what the right approach to this is. This seemed like a reasonable option to me. The only other option that I could think of would be to require that both a network interface and a hostname/IP be provided. But for a clustered environment, that won't work. So picking the first one seemed the best option to me. Any other suggestion?